app/script/view_model/content/InputBarViewModel.js

```diff
@@ -461,34 +480,13 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
 
   sendMessage(messageText) {
     if (messageText.length) {
-      const participatingUserEntities = this.conversationEntity().participating_user_ets();
-      const mentionEntities = this.parseForMentions(messageText, participatingUserEntities);
-      this.conversationRepository.sendTextWithLinkPreview(messageText, this.conversationEntity(), mentionEntities);
+      const mentionEntities = this.currentMentions;
+      this.conversationRepository
+        .sendTextWithLinkPreview(messageText, this.conversationEntity(), mentionEntities)
+        .then(() => (this.currentMentions = []));
```

Comment:
No reason to wait for the then. If something fails, mentions will not be unset.

````
this.conversationRepository.sendTextWithLinkPreview(messageText, this.conversationEntity(), this.currentMentions);
this.currentMentions = [];
````

Reply:
I would argue that they could even be unset at the same position as the input is set to `''` in `onInputEnter`.

Reply:
@gregor @AndyLnd fair points. I think I'll go for the `onInputEnter` solution :+1: